src/measure_theory/measure_space.lean

```diff
@@ -2218,6 +2218,65 @@ end
 
 end is_complete
 
+namespace measure_theory
+
+lemma outer_measure.to_measure_zero [measurable_space α] : (0 : outer_measure α).to_measure
+  ((le_top).trans outer_measure.zero_caratheodory.symm.le) = 0 :=
+by rw [← measure.measure_univ_eq_zero, to_measure_apply _ _ measurable_set.univ,
+  outer_measure.coe_zero, pi.zero_apply]
+
+section trim
+
+/-- Restriction of a measure to a sub-sigma algebra.
+It is common to see a measure `μ` on a measurable space structure `m0` as being also a measure on
+any `m ≤ m0`. Since measures in mathlib have to be trimmed to the measurable space, `μ` itself
+cannot be a measure on `m`, hence the definition of `μ.trim hm`. -/
+def measure.trim {m m0 : measurable_space α} (μ : @measure α m0) (hm : m ≤ m0) : @measure α m :=
```

Comment:
Does it make sense to reference `measure_theory.outer_measure.trim` in the docstring?

Reply:
I am unsure about the name `measure.trim` for the new definition. I would have chosen `restrict`, but that's taken. `trim` is indeed used for outer measures and I am not 100% sure of the relation between my new def and the existing notion.
I need to try and prove a few results relating those notions, in order to get a better picture.

If anybody has an idea for another name, any suggestion is welcome!

Reply:
I added a new lemma `to_outer_measure_trim_eq_trim_to_outer_measure` which illustrate the relation between the two trim notions. It looks like the name `trim` is the right one.